src/chainrules.jl

```diff
@@ -0,0 +1,152 @@
+# ffts
+function ChainRulesCore.frule((_, Δx, _), ::typeof(fft), x::AbstractArray, dims)
+    y = fft(x, dims)
+    Δy = fft(Δx, dims)
+    return y, Δy
+end
+function ChainRulesCore.rrule(::typeof(fft), x::AbstractArray, dims)
```

Comment:
Since `fft(x)` just calls `plan_fft(x) * x`, shouldn't the chain rules be defined for `mul!` operations with a `Plan`?

Reply:
I thought about this but then I noticed the following problem: the interface does not allow to dispatch on something like `Plan{fft}` or `Plan{rfft}`, so we don't know which transform the plan computes. In the reverse-mode rules this would be desirable though (if we don't want to use heuristics based on e.g. sizes of `x` and `P * x`) since the rules for `rfft`/`irfft`/`brfft` require some additional scaling (caused by the conjugate symmetry). (The other problem with `P.region` that I mentioned in the comment above seems easier to solve, we could just add a `region(::Plan)` function that falls back to the field.)